### PR TITLE
Improvements to the integration test screenshot service

### PR DESCRIPTION
**Pull request by sharwell**
* Avoid concurrency
* Capture a `BitmapSource` instead of a `Bitmap` (errors reported during save are more actionable on this code path)
* Make `ScreenshotService` static

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs [hunk 1]
```diff
@@ -4,20 +4,30 @@
...
 using System.Windows.Forms;
+using System.Windows.Media.Imaging;
+using PixelFormats = System.Windows.Media.PixelFormats;
 
 namespace Microsoft.VisualStudio.IntegrationTest.Utilities
 {
-    internal class ScreenshotService
+    internal static class ScreenshotService
     {
+        private static readonly object _gate = new object();
```
Comment (jaredpar) [inline src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs]: `s_gate`

File: src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs [hunk 2]
```diff
@@ -26,7 +36,12 @@ public static void TakeScreenshot(string fullPath)
                 var directory = Path.GetDirectoryName(fullPath);
                 Directory.CreateDirectory(directory);
 
-                bitmap.Save(fullPath, ImageFormat.Png);
+                using (var fileStream = new FileStream(fullPath, FileMode.Create))
```
Comment (jaredpar) [inline src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs]: Nit: use `FileAccess.Write` vs the implicit `FileAccess.ReadWrite` that is currently present.

**Comment by sharwell** [comment]
> Synchronization code looks good. Out of my depth reviewing the Bitmap portion of the change.

Based on the failures in CI, I can confirm it at minimum *works* 😆

File: src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs]: ➡️ Amended first commit to fix

File: src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs (hunk 2 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestUtilities/ScreenshotService.cs]: ➡️ Amended second commit to fix

**Comment by sharwell** [comment]
@dpoeschl Can you review the use of `BitmapSource` (second commit)?

**Comment by sharwell** [comment]
retest windows_coreclr_debug_prtest please

**Comment by sharwell** [comment]
retest windows_release_unit32_prtest please

**Comment by jinujoseph** [comment]
@sharwell this needs to be retargetted

**Comment by sharwell** [comment]
This is now retargeted. @ivanbasov or @mavasani  can I get a review of this?

**Pull request merged**